src/core/server/saved_objects/migrations/model/retry_state.ts

```diff
@@ -10,20 +10,27 @@ import { State } from '../state';
 
 export const delayRetryState = <S extends State>(
   state: S,
-  errorMessage: string,
+  errorMessage: string, // comes in as "[index_not_yellow_timeout] Timeout waiting for the status of the [${index}] index to become 'yellow'"
   /** How many times to retry a step that fails */
-  maxRetryAttempts: number
+  maxRetryAttempts: number,
+  /** optional link to docs */
+  docLink?: string
```

Comment:
Letting `delayRetryState` construct the final message with the docLinks makes it hard to provide anchor links to the correct section which isn't something we have to do in this PR, but could be a valuable improvement as that page grows in size.

I think it would be better if:
1. like it currently works, the action returns `"[index_not_yellow_timeout] Timeout waiting for the status of the [${index}] index to become 'yellow'"`
2. Then the model calls `delayRetryState` with `errorMessage: "[index_not_yellow_timeout] Timeout waiting for the status of the [${index}] index to become 'yellow'". Refer to ${docLink} for information on how to resolve the issue.` The model knows exactly what went wrong and during what step, so in the future we will tweak the error message to e.g. `${docLink#some_specific_error_anchor}`.
3. `delayRetryState` just logs the errorMessage as is for every retry.
4. When retryCount > maxRetryAttempts we can improve the current logs with `reason: Unable to complete the ${state.controlState} step after ${maxRetryAttempts} attempts, terminating. The last failure message was: ${errorMessage}`

This would mean we give a little bit more context and the docLink in the last fatal error which is the most likely place a user will look.

Reply:
We're thinking along the same lines, only I was going to do the refactor and provide the anchor links in a follow-up PR, after this one that adds the new docs' sections was merged.

It makes perfect sense to do that now though, then the follow-up should be low-hanging fruit.

Reply:
I went back and forth on extracting the error-message construction in [`2e8dc9a` (#130352)](https://github.com/elastic/kibana/pull/130352/commits/2e8dc9a0c99c501afa76f7fec87078ac354acd1e) into a helper function but decided to leave these as-is directly in the model. To me, it's easier to read and start changing anchor links. If we end up using the same anchor link in more than one place, I'll extract the message construction into helper functions, similar to [`fatalReasonDocumentExceedsMaxBatchSizeBytes`](https://github.com/elastic/kibana/blob/main/src/core/server/saved_objects/migrations/model/model.ts#L37-L46)

If we end up using the same sublink

Reply:
> I think it'll be easier to read and add anchor links.

I like your thinking :+1: